File: Csharp/OOP/CustomerApp/CustomerApp/Customer.cs
using System;

namespace CustomerApp
{
    class Customer
    {
        private int _id;
        private string _firstname, _lastname;
        private static int _nextid;

        static Customer(){
            _nextid = 1000;
            }

        public Customer(string firstname,string lastname) {
            _firstname = firstname;
            _lastname = lastname;
            _nextid += 1;
            _id = _nextid;
        }

        public int Id {
            get { return _id; }
        }
        public string FirstName {
            get { return _firstname; }
        }
        public string LastName {
            get { return _lastname; }
        }
    }
}

File: Nodejs/Typescript/Customer.ts
export class Customer{
    constructor(private id:number,
                private firstname:string,
                private lastname:string){}
    get Id(){
        return this.id;
    }
    get Fullname(){
        return this.firstname+" "+this.lastname;
    }
}
export class Address{
    constructor(private streetname:string,
                private cityname:string,
                private zipcode:number){}
    get Fulladdress(){
        return this.streetname+" "+this.cityname+" "+this.zipcode;
    }
}